src/scr.c

```diff
@@ -37,6 +37,19 @@
 #include "dtcmp.h"
 #endif /* HAVE_LIBDTCMP */
 
+/* define which state we're in for API calls, this is to help ensure
+ * users call SCR functions in the correct order */
+typedef enum {
+    SCR_STATE_UNINIT,     /* before init and after finalize */
+    SCR_STATE_INIT,       /* between init/finalize */
+    SCR_STATE_RESTART,    /* between start/complete restart */
+    SCR_STATE_CHECKPOINT, /* between start/complete checkpoint */
```

Comment:
We had talked about keeping Checkpoint around as a convenience and/or backwards compatibility.  I'll leave this state in here until we remove the API calls.